### PR TITLE
DDB-1021 - Measure upload - edit existing values

### PR DESCRIPTION
**Pull request by leighjohnson-friday**
Allow users to update existing measure values

### CODE REVIEW

File: helpers/measures.js [hunk 1]
```diff
@@ -59,9 +63,82 @@ const getCategory = async (name) => {
   return category;
 }
 
+const getMeasureEntitiesFromGroup = (groupEntities, metricId) => {
+  const measureEntities = groupEntities.filter(entity => entity.metricID === metricId)
+  const sortedEntities = measureEntities.sort((a, b) => moment(a.date, 'DD/MM/YYYY').valueOf() - moment(b.date, 'DD/MM/YYYY').valueOf());
+  return sortedEntities;
+}
+
+const removeBlankEntityInputValues = (entityInputs) => {
```
Comment (dtaylor7) [inline helpers/measures.js:72]: we have a rathe horrible function for this already if you want to use https://github.com/alphagov/co-eu-transition-dashboard/blob/bdeeea0193288fa71626a4fe33b13615794cc84d/helpers/utils.js#L1

File: pages/data-entry-entity/measure-value/MeasureValue.js [hunk 2]
```diff
@@ -104,85 +239,97 @@ class MeasureValue extends Page {
...
       entities: {},
-      date: selecteDate.split('/')
-    }
+      date: selecteDate.split("/"),
+    };
 
-    measureForSelectedDate.forEach(measure => {
-      uiInputs.forEach(uiInput => {
-        // label and label2 are generated from applyLabelToEntities using filter, filet2, filterValue, filterValue2
+    measureForSelectedDate.forEach((measure) => {
```
Comment (dtaylor7) [inline pages/data-entry-entity/measure-value/MeasureValue.js:284]: looks like the same stuff on line 80?

**Review by dtaylor7 — APPROVED** [review]
more great work on the reuse. few optional comments.

a lot of code has moved around, suggest hiten to fully re-test add and edit.

👍

File: helpers/measures.js (hunk 1 above)
Comment (leighjohnson-friday) [reply helpers/measures.js:72]: Ah nice, will take a look

**Pull request merged**